erpnext/projects/doctype/project/project.py

```diff
@@ -38,6 +39,17 @@ def validate(self):
 		self.send_welcome_email()
 		self.update_costing()
 		self.update_percent_complete()
+		self.validate_task()
+
+	def validate_task(self):
+		tasks = frappe.get_all("Task", {"project":self.project_name})
+		for task in tasks:
+			doc = frappe.get_doc("Task", task.name)
+			if self.expected_start_date:
+				doc.exp_start_date = self.expected_start_date
+			if self.expected_end_date:
+				doc.exp_end_date = self.expected_end_date
```

Comment:
Do we want to keep overriding the dates whenever a project is saved?

Reply:
we need to override it when the date is changed.

Reply:
I dont think. it should not override on project save. it should trigger when project start date and end date change